packages/material-ui-styles/src/index.d.ts

```diff
@@ -199,15 +198,12 @@ declare module '@material-ui/styles/StylesProvider' {
 }
 
 declare module '@material-ui/styles/ThemeProvider' {
-  import { Theme } from '@material-ui/core';
-
-  const ThemeContext: React.Context<Theme>;
-
-  export interface ThemeProviderProps {
+  export interface ThemeProviderProps<Theme> {
     children: React.ReactNode;
     theme: Theme | ((outerTheme: Theme) => Theme);
```

Comment:
This assumes that the outer theme has the same type. We need make the potential difference explicit so that users get an implicit any.